utilities/blob_db/blob_db_impl.cc

```diff
@@ -867,15 +867,17 @@ Status BlobDBImpl::SingleDelete(const WriteOptions& wopts,
 }
 
 Status BlobDBImpl::Write(const WriteOptions& opts, WriteBatch* updates) {
-  class Handler1 : public WriteBatch::Handler {
+  class BlobInserter : public WriteBatch::Handler {
    public:
-    explicit Handler1(BlobDBImpl* i) : impl(i), previous_put(false) {}
+    explicit BlobInserter(BlobDBImpl* i, SequenceNumber seq)
+        : impl(i), sequence(seq), has_put(false) {}
 
     BlobDBImpl* impl;
+    SequenceNumber sequence;
     WriteBatch updates_blob;
     Status batch_rewrite_status;
     std::shared_ptr<BlobFile> last_file;
-    bool previous_put;
+    bool has_put;
```

Comment:
What's the need for exposing all these fields as public? I think you can hide almost all of them.
It seems like you only access batch_rewrite_status externally in line 976. So may be you could add another function, say status() to return just batch_rewrite_status value.